nvflare/private/fed/utils/fed_utils.py

```diff
@@ -198,12 +202,48 @@ def get_scope_info():
         return [], "processing_error"
 
 
-def fobs_initialize():
+def fobs_initialize(workspace: Workspace = None, job_id: str = None):
+    nvflare_fobs_initialize()
+
+    custom_fobs_initialize(workspace, job_id)
+
+
+def custom_fobs_initialize(workspace: Workspace = None, job_id: str = None):
+    if workspace:
+        site_custom_dir = workspace.get_client_custom_dir()
+        decomposer_dir = os.path.join(site_custom_dir, FOBS_DECOMPOSER_DIR)
+        if os.path.exists(decomposer_dir):
+            register_custom_folder(decomposer_dir)
+
+        if job_id:
+            app_custom_dir = workspace.get_app_config_dir(job_id)
+            decomposer_dir = os.path.join(app_custom_dir, FOBS_DECOMPOSER_DIR)
+            if os.path.exists(decomposer_dir):
+                register_custom_folder(decomposer_dir)
+
+
+def nvflare_fobs_initialize():
     flare_decomposers.register()
     common_decomposers.register()
     private_decomposers.register()
 
 
+def register_ext_decomposers(decomposer_module: str):
```

Comment:
The "decomposer_module" could also be a list of strings. For example, customer may use other frameworks that all have their own separate decomposers.

Reply:
made changes to support both str and List[str] cases.